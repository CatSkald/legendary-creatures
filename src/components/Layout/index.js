import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
