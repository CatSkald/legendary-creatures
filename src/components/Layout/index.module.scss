.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
