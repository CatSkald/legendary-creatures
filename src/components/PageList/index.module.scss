.pageList {
  margin-bottom: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    display: grid;
    grid-gap: var(--space-sm);
    grid-template-columns: repeat(
      auto-fit,
      minmax(250px, calc(33% - var(--space-sm)))
    );
  }
}
