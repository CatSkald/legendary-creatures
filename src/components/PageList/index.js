import { pageList } from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";
import CreaturePage from "../CreaturePage";
import { LocaleContext } from "../Layout";
import { getCreaturesPath } from "../../utils/url-helpers";

const PageList = (props) => {
  const { language } = React.useContext(LocaleContext);

  return (
    <section className={pageList}>
      {props.pages.map(
        ({
          node: {
            frontmatter: { date, description, title, image },
            fields: { slug },
          },
        }) => (
          <CreaturePage
            key={slug}
            slug={`${getCreaturesPath(language.code)}/${slug}`}
            date={date}
            title={title}
            description={description}
            image={image}
          />
        ),
      )}
    </section>
  );
};

PageList.propTypes = {
  pages: PropTypes.array.isRequired,
};

export default PageList;
