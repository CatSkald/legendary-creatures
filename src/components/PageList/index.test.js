import React from "react";
import * as Gatsby from "gatsby";
import { render } from "@testing-library/react";
import each from "jest-each";

import { LocaleContext } from "../Layout";
import PageList from "../PageList";

function createPage() {
  return {
    node: {
      frontmatter: {
        date: new Date().toString(),
        description: "test description",
        title: "My Title",
        image: "",
      },
      fields: { slug: "/" },
    },
  };
}

beforeEach(() => {
  jest.clearAllMocks();
});

describe("PageList", () => {
  const createLocaleContext = (language) => ({
    language: {
      code: language,
    },
  });

  each(["en", "uk", "ru"]).it(
    "renders correct props for %s",
    (languageCode) => {
      const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
      useStaticQuery.mockImplementation(() => ({
        listImages: {
          edges: [
            {
              node: {
                childImageSharp: {
                  metadata: { relativePath: "cover.jpg" },
                },
              },
            },
          ],
        },
      }));

      const { container } = render(
        <LocaleContext.Provider value={createLocaleContext(languageCode)}>
          <PageList pages={[createPage()]} />
        </LocaleContext.Provider>,
      );
      expect(container).toMatchSnapshot();
    },
  );
});
