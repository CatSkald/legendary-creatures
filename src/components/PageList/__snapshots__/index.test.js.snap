// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageList renders correct props for en 1`] = `
<div>
  <section
    class="pageList"
  >
    <a
      class="creaturePage__link"
      href="undefined/creatures//"
      hreflang="en"
    >
      <section
        class="creaturePage"
      >
        <div
          class="creaturePage__content"
        >
          <h3
            class="creaturePage__title"
          >
            My Title
          </h3>
          <p
            class="creaturePage__description"
          >
            test description
          </p>
        </div>
      </section>
    </a>
  </section>
</div>
`;

exports[`PageList renders correct props for ru 1`] = `
<div>
  <section
    class="pageList"
  >
    <a
      class="creaturePage__link"
      href="undefined/существа//"
      hreflang="ru"
    >
      <section
        class="creaturePage"
      >
        <div
          class="creaturePage__content"
        >
          <h3
            class="creaturePage__title"
          >
            My Title
          </h3>
          <p
            class="creaturePage__description"
          >
            test description
          </p>
        </div>
      </section>
    </a>
  </section>
</div>
`;

exports[`PageList renders correct props for uk 1`] = `
<div>
  <section
    class="pageList"
  >
    <a
      class="creaturePage__link"
      href="undefined/істоти//"
      hreflang="uk"
    >
      <section
        class="creaturePage"
      >
        <div
          class="creaturePage__content"
        >
          <h3
            class="creaturePage__title"
          >
            My Title
          </h3>
          <p
            class="creaturePage__description"
          >
            test description
          </p>
        </div>
      </section>
    </a>
  </section>
</div>
`;
